Jed/projection

jeremylt opened the pull request:
WIP - Added command line flag to add quadrature points but it seems to have no effect on error

jeremylt commented:
ran against "test-kspsolve -op_type project2affine -M 64,64,64 -add_quad_pts 2 -ksp_monitor -pc_type jacobi -ksp_type cg -log_view" and saw the approximate times mentioned in slack

jedbrown commented on finite-element/test/t120-poissonksp.sh:
```diff
@@ -1,56 +0,0 @@
-#!/bin/sh
```
Why is this file deleted? Seems to me we don't want to break this feature.

jeremylt replied:
The output of my KSP was slightly different - same information organized differently, so I temporarily pulled the file. I'll put it back.

jedbrown commented on hpgmgTest/Makefile:
```diff
@@ -0,0 +1,14 @@
+HPGMG_ARCH = hpgmgTest
```
These files shouldn't be added to the project -- they aren't relevant to other people.

jedbrown commented on hpgmgTest/reconfigure-hpgmgTest.py:
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/python
```
Same with this file.

jeremylt replied:
I'll fix both of those.

jedbrown reviewed (changes requested):
The commits should probably be squashed to avoid adding a bunch of binary files and then deleting them. The general implementation strategy looks fine to me. It would help to add an automated test of the new higher order quadrature. We can optimize in a separate pass.

Commit:
Added flag for additional quadrature points

jeremylt commented:
I squashed the commits. I also modified the implementation slightly to ensure that OpCreateFromOptions built tensors with the correct number of quadrature points. The test is currently omitted, per our discussion last night.
